controllers/registeredservice_controller.go

```diff
@@ -109,13 +109,27 @@ func (r *RegisteredServiceReconciler) addServiceToCatalogs(ctx context.Context,
 
 	var errs []error
 	for _, sc := range catalogs.Items {
-		if rs.Spec.Constraints == nil || envtag.Match(sc.Name, rs.Spec.Constraints.Environments) {
+		constraints_match := true
+		if rs.Spec.Constraints != nil {
+			constraints_match = envtag.Match(sc.Name, rs.Spec.Constraints.Environments)
+		}
+
+		if constraints_match {
```

Comment:
if you like it you can add for RegisteredService a method like [we did for ServiceClasses](https://github.com/primaza/primaza/blob/966880a5c65af36746d14d0bbec18a77da56f7b7/api/v1alpha1/serviceclass_types.go#LL89C1-L95C2) and refactor these lines as follows:

```suggestion
		if envtag.Match(sc.Name, rs.Spec.GetEnvironmentConstraints()) {
```

Reply:
I like it, done